modules/setting/setting.go

```diff
@@ -643,6 +631,21 @@ please consider changing to GITEA_CUSTOM`)
 	Domain = sec.Key("DOMAIN").MustString("localhost")
 	HTTPAddr = sec.Key("HTTP_ADDR").MustString("0.0.0.0")
 	HTTPPort = sec.Key("HTTP_PORT").MustString("3000")
+
+	defaultAppURL := string(Protocol) + "://" + Domain + ":" + HTTPPort + "/"
```

Comment:
I think it should not add default ports to url depending on protocol (https 443 and http 80)